src/main/java/org/skriptlang/skript/lang/util/SkriptQueue.java

```diff
@@ -0,0 +1,85 @@
+package org.skriptlang.skript.lang.util;
+
+import ch.njol.yggdrasil.YggdrasilSerializable;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+
+/**
+ * A queue of elements.
+ * This is designed not to support null elements, i.e. putting in nothing should do nothing.
+ */
+public class SkriptQueue extends LinkedList<@NotNull Object>
+	implements Deque<Object>, Queue<Object>, YggdrasilSerializable {
+
+	@Override
+	public boolean add(Object element) {
+		if (element == null)
+			return false;
+		return super.add(element);
+	}
+
+	@Override
+	public void add(int index, Object element) {
+		if (element != null)
+			super.add(index, element);
+	}
+
+	@Override
+	public void addFirst(Object element) {
+		if (element != null)
+			super.addFirst(element);
+	}
+
+	@Override
+	public void addLast(Object element) {
+		if (element != null)
+			super.addLast(element);
+	}
+
+	@Override
+	public boolean contains(Object o) {
+		if (o == null)
+			return false;
+		return super.contains(o);
+	}
+
+	@Override
+	public Object set(int index, Object element) {
+		if (element == null)
+			return null;
+		return super.set(index, element);
+	}
+
+	@Override
+	public boolean addAll(int index, Collection<?> list) {
+		List<?> copy = new ArrayList<>(list);
+		copy.removeIf(Objects::isNull);
+		return super.addAll(index, copy);
+	}
+
+	@Override
+	public @NotNull Object @NotNull [] toArray() {
+		return super.toArray();
+	}
+
+	public Object removeSafely(int i) {
```

Comment:
This isn't nullable, since if you follow the contract it will never be null.
Nullable is for things that may be null for reasons outside of the caller's control

Reply:
Might we instead prefer a Preconditions check instead of these null returns?

Reply:
No, the point of "safely" is that it doesn't throw an error. Preconditions throws an error. How would this help?